### PR TITLE
plumb through an ignore_snooze and use it in user initiated actions

### PR DESCRIPTION
**Pull request by songgao**
Related: https://github.com/keybase/keybase/pull/3022

### CODE REVIEW

**Comment by songgao** [comment]
bump!

File: keybase/context.go [hunk 1]
```diff
@@ -74,8 +74,8 @@ func newContextCheckCmd(cfg Config, log Log, isCheckCommand bool) *context {
 }
 
 // NewUpdaterContext returns an updater context for Keybase
-func NewUpdaterContext(appName string, pathToKeybase string, log Log, isCheck bool) (updater.Context, *updater.Updater) {
-	cfg, err := newConfig(appName, pathToKeybase, log)
+func NewUpdaterContext(appName string, pathToKeybase string, log Log, isCheck bool, ignoreSnooze bool) (updater.Context, *updater.Updater) {
```
Comment (maxtaco) [inline keybase/context.go]: Not ideal to have a bunch of bools piling up on top of each other here. It's quite easy to mix them up. Do all 4 combinations make sense?

File: service/main.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func run(f flags) {
 		svc.Run()
 	case "clean":
 		if runtime.GOOS == "windows" {
-			ctx, _ := keybase.NewUpdaterContext(f.appName, f.pathToKeybase, ulog, true)
+			ctx, _ := keybase.NewUpdaterContext(f.appName, f.pathToKeybase, ulog, true, false)
```
Comment (maxtaco) [inline service/main.go]: this sort of thing isn't great. `foo(true, false, true, true)`....

**Comment by songgao** [comment]
Thanks @maxtaco; ready for another look!

**Comment by songgao** [comment]
That’s one. Also people have complained that `keybase update check` doesn’t pull updates when it’s snoozed, so this should fix that too.

**Pull request merged**